File: SDK-sources/OutbrainSDK/AdIds/OBAppleAdIdUtil.m
//
//  OBAppleAdIdUtil.m
//  OutbrainSDK
//
//  Created by Oded Regev on 10/2/14.
//  Copyright (c) 2014 Outbrain. All rights reserved.
//

#import "OBUtils.h"
#import "OutbrainManager.h"
#import "OBAppleAdIdUtil.h"
#import <AdSupport/AdSupport.h>
#import <AppTrackingTransparency/AppTrackingTransparency.h>

@implementation OBAppleAdIdUtil

+ (BOOL)isOptedOut {
    if (@available(iOS 14, *)) {
        return [ATTrackingManager trackingAuthorizationStatus] != ATTrackingManagerAuthorizationStatusAuthorized;
    }
    else { // for devices with iOS < 14.0
        return ![[ASIdentifierManager sharedManager] isAdvertisingTrackingEnabled];
    }
}

+ (NSString *)getAdvertiserId {
    if ([OutbrainManager sharedInstance].testMode &&
        [[OBUtils deviceModel] isEqualToString:@"Simulator"])
    {
        return @"F22700D5-1D49-42CC-A183-F3676526035F"; // dev hack because simulator returns 0000-0000-0000-0000
    }
    return [[[ASIdentifierManager sharedManager] advertisingIdentifier] UUIDString];
}


@end

File: SwiftSDK-sources/OutbrainSDK/Utils/OBAppleAdIdUtil.swift
//
//  OBAppleAdIdUtil.swift
//  OutbrainSDK
//
//  Created by Dror Seltzer on 20/06/2023.
//

import Foundation
import AdSupport
import AppTrackingTransparency

class OBAppleAdIdUtil {
    
    // check if user has opted out of tracking
    static var isOptedOut: Bool {
        if Outbrain.testMode { return false }
        
        return ATTrackingManager.trackingAuthorizationStatus != .authorized
    }

    
    // get the apple advertising id
    static var advertiserId: String {
        if Outbrain.testMode {
            return "F22700D5-1D49-42CC-A183-F3676526035F" // dev hack because simulator returns 0000-0000-0000-0000
        }
        
        let idfa = ASIdentifierManager.shared().advertisingIdentifier.uuidString
        
        guard !idfa.isEmpty else { return "null" }
        return idfa
    }
}
